module2/module2.js

```diff
@@ -0,0 +1,196 @@
+// ==== Module#2 Functions ====
+mocha.setup('bdd');
+const expect = chai.expect;
+
+// ==== Currying ====
+
+function curry(f) {
+  return function(a) {
+    return function(b) {
+      return function(c) {
+        return function(d) {
+          return f(a, b, c, d);
+        };
+      };
+    };
+  };
+};
+
+function joinFunction(...args) {
+  return args.join(' ');
+}
+
+const mergeWords =  curry(joinFunction);
+
+describe('mergeWords', () => {
+  it('should merge words into sentence', () => {
+    const result = mergeWords('GNU')('is')('not')('Unix.'); //removed function call, is it ok?
+
+    expect(result).to.equal('GNU is not Unix.');
+  });  
+});
+
+// ==== Every/Some ====
+const goodUsers = [
+    { id: 1 },
+    { id: 2 },
+    { id: 3 },
+];
+
+// `checkUsersValid` is the function you'll define
+const checkUsersValid = (goodUsers) => {
+  return function(usersToTest) {
+    let absentUsers = usersToTest.filter( (user) => filterAbsentUser(user, goodUsers))
+    return absentUsers.length === 0; 
+  };
+};
+
+function filterAbsentUser(user, goodUsers) {
+  if (goodUsers.some(e => e.id === user.id)) {
+    return false;
+  }
+  return true;
+}
+
+const testAllValid = checkUsersValid(goodUsers);
+
+
+describe('checkUsersValid', () => {
+  it('#case 1', () => {
+    const result = testAllValid([
+      { id: 2 },
+      { id: 1 },
+    ]);
+
+    expect(result).to.equal(true);
+  });
+  it('#case 2', () => {
+    const result = testAllValid([
+      { id: 2 },
+      { id: 4 },
+      { id: 1 },
+    ]);
+
+    expect(result).to.equal(false);
+  });
+});
+
+// ==== Reduce ====
+const reducer = (accumulator, currentValue) => { 
+  if(accumulator[currentValue] === undefined) {
+    accumulator[currentValue] = 1;
```

Comment:
I would update it

`accumulator[currentValue] = 0;`
`// return accumulator;`  <==== in my opinion it's better to have only one return from a function if possible.